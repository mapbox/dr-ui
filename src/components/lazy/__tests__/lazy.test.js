import renderer from 'react-test-renderer';
import { testCases } from './lazy-test-cases.js';

describe('Lazy', () => {
  describe(testCases.basic.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.larger.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.larger;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
