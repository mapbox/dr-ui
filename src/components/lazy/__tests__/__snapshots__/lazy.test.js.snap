// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lazy A larger example renders as expected 1`] = `
<div
  className="relative"
  style={
    Object {
      "height": 850,
    }
  }
>
  <div
    className="flex-parent flex-parent--center-cross flex-parent--center-main absolute top right bottom left bg-darken10 z5 round"
  >
    <div
      className="flex-child loading loading--s"
    />
  </div>
</div>
`;

exports[`Lazy Basic renders as expected 1`] = `
<div
  className="relative h30-mm h36"
  style={Object {}}
>
  <div
    className="flex-parent flex-parent--center-cross flex-parent--center-main absolute top right bottom left bg-darken10 z5 round"
  >
    <div
      className="flex-child loading loading--s"
    />
  </div>
</div>
`;
